addons/calendar/static/src/views/attendee_calendar/attendee_calendar_model.js

```diff
@@ -77,6 +77,17 @@ export class AttendeeCalendarModel extends CalendarModel {
         );
         const currentPartnerId = this.user.partnerId;
         if (!isEveryoneFilterActive) {
+            const statusDomain = this.env.searchModel._domain.filter((leaf) => Array.isArray(leaf) && leaf.includes("attendee_status"));
+            let hasSearchStateList = false;
+            let searchStateList = [];
+            if (statusDomain.length && statusDomain[0].length === 3) {
+                hasSearchStateList = true;
+                searchStateList = await this.orm.call(
+                    "calendar.event",
+                    "get_search_attendee_status",
+                    [statusDomain[0][1], statusDomain[0][2]],
+                );
+            }
```

Comment:
Cannot we use `removeDomainLeaves` function to extract the domain part we need to search what you want? 🧐 
https://github.com/odoo/odoo/blob/d1f450adde7fba78fe1274244b31deeac430467c/addons/web/static/src/core/domain.js#L72-L128

Reply:
I want [filter_domain_leaf](https://github.com/odoo/odoo/blob/master/addons/resource/models/utils.py#L55) kind of method in js to filter specific domain leaf :innocent: 
Should I try to add it? If yes then where in calendar or web? :thinking: 

OR should I go with removing All key except  `attendee_status` via above method?

Reply:
For the moment, we will refactor that function (see task-3299357) and after that we will replace `removeDomainLeaves` by this one once `filter_domain_leaf` will be converted in JS. 🙂

Reply:
So, for the moment, I would say, you can use removeDomainLeaves since you have all the fields in the model (I guess?) and once we will have `filter_domain_leaf`, we will change your code to use that function instead. 🙂 

I would be easier for you to not wait too long to have that function. 🙂

Reply:
Hello @xavierbol and @arj-odoo  :wave: 

I tried `removeDomainLeaves` but it'll not help in optimising code because I need operator and value of domainLeaf if operand is `attendee_status` to get proper attendee statuses list according  search result.

So I optimize my code to get proper result when domain has more then one leaf which contains `attendee_state`